ops/testing.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Infrastructure to build unit tests for charms using the ops library."""
+
+from ._private.harness import (
+    ActionFailed,
+    ActionOutput,
+    AppUnitOrName,
+    CharmBase,
+    CharmMeta,
+    CharmType,
+    Container,
+    ExecArgs,
+    ExecHandler,
+    ExecProcess,
+    ExecResult,
+    Harness,
+    ReadableBuffer,
+    RelationNotFoundError,
+    RelationRole,
+    YAMLStringOrFile,
+    charm,
+    framework,
+    model,
+    pebble,
+    storage,
+)
+
+# The Harness testing framework.
+_ = ActionFailed
```

Comment:
What's the difference/advantage of doing this over `__all__`?

Reply:
`__all__` causes Sphinx to get very confused, and I couldn't figure out any way to keep `__all__` and de-confuse it.